mne/preprocessing/eyetracking/eyetracking.py

```diff
@@ -164,3 +170,162 @@ def _convert_mm_to_m(array):
 
 def _convert_deg_to_rad(array):
     return array * np.pi / 180.0
+
+
+def convert_units(inst, calibration, to="radians"):
+    """Convert Eyegaze data from pixels to radians of visual angle or vice versa.
+
+    .. warning::
+        Currently, depending on the units (pixels or radians), eyegaze channels may not
+        be reported correctly in visualization functions like :meth:`mne.io.Raw.plot`.
+        They will be shown  correctly in :func:`mne.viz.eyetracking.plot_gaze`.
+        See :gh:`11879` for more information.
+
+    .. Important::
+       There are important considerations to keep in mind when using this function,
+       see the Notes section below.
+
+    Parameters
+    ----------
+    inst : instance of Raw, Epochs, or Evoked
+        The Raw, Epochs, or Evoked instance with eyegaze channels.
+    calibration : Calibration
+        Instance of  Calibration, containing information about the screen size
+        (in meters), viewing distance (in meters), and the screen resolution
+        (in pixels).
+    to : str
+        Must be either ``"radians"`` or ``"pixels"``, indicating the desired unit.
+
+    Returns
+    -------
+    inst : instance of Raw | Epochs | Evoked
+        The Raw, Epochs, or Evoked instance, modified in place.
+
+    Notes
+    -----
+    There are at least two important considerations to keep in mind when using this
+    function:
+
+    1. Converting between on-screen pixels and visual angle is not a linear
+       transformation. If the visual angle subtends less than approximately ``.44``
+       radians (``25`` degrees), the conversion could be considered to be approximately
+       linear. However, as the visual angle increases, the conversion becomes
+       increasingly non-linear. This may lead to unexpected results after converting
+       between pixels and visual angle.
+
+    * This function assumes that the head is fixed in place and aligned with the center
+      of the screen, such that gaze to the center of the screen results in a visual
+      angle of ``0`` radians.
+
+    .. versionadded:: 1.7
+    """
+    _validate_type(inst, (BaseRaw, BaseEpochs, Evoked), "inst")
+    _validate_type(calibration, Calibration, "calibration")
+    _check_option("to", to, ("radians", "pixels"))
+    _check_calibration(calibration)
+
+    # get screen parameters
+    screen_size = calibration["screen_size"]
+    screen_resolution = calibration["screen_resolution"]
+    dist = calibration["screen_distance"]
+
+    # loop through channels and convert units
+    converted_chs = []
+    for ch_dict in inst.info["chs"]:
+        if ch_dict["coil_type"] != FIFF.FIFFV_COIL_EYETRACK_POS:
+            continue
```

Comment:
@scott-huberty this is the other change I made -- `if not x: continue` rather than `if x: <indent everything another 4 spaces`. I find it the same readability but less indented which is nice

Reply:
Good idea thx!